lib/nerves_wpa_supplicant.ex

```diff
@@ -21,21 +21,67 @@ defmodule Nerves.WpaSupplicant do
 
   defstruct port: nil,
             ifname: nil,
+            restart: :transient,
+            control_socket_path: "",
             requests: []
 
   @doc """
   Start and link a Nerves.WpaSupplicant that uses the specified control
   socket.
   """
-  def start_link(ifname, control_socket_path, opts \\ []) do
-    GenServer.start_link(__MODULE__, {ifname, control_socket_path}, opts)
+  def start_link(ifname, control_socket_path, restart \\ :transient, opts \\ []) do
+    GenServer.start_link(__MODULE__, {ifname, control_socket_path, restart}, opts)
+  end
+  def start_link([ifname, control_socket_path, restart, opts]) do
```

Comment:
Why do we need so many cases for starting the module? Can we not just have one, and subsequently only one init as well?

Reply:
This is due to the couple facts:
1.  multiple ways of starting a supervisor - I went for Supervisor.child_spec(...) hence the start_link([ array items ]) clause
2. There's another Nerves.Network module (WiFiManager) that uses this module and I wanted to keep backwards compatibility - hence restart \\ :transient that is one item we introduced

Reply:
Can you not just keep them all in one way then, even if it is the old way? Just so that it is consistent